### ISSUE TITLE
Training speed

### ISSUE THREAD

**Issue by yy18dlz** [open-mmlab/mmdetection#4394]
Hi, This is awesome work！I love lt！
But, I have some problem....
I found the taining speed may not as faster as i expected

here is my use analyze_logs.py:
slowest epoch 49, average time is 1.1886
fastest epoch 11, average time is 0.9775
time std over epochs is 0.0323
average iter time: 1.0163 s/iter

this means for one epoch it's need about 4 hours....
the same net on darknet only need about 1 hours for one epoch
Is there any good suggestion here?

log
{"mode": "train", "epoch": 161, "iter": 100, "lr": 0.001, "memory": 5572, "data_time": 0.19335, "loss_cls": 141.72443, "loss_conf": 499.94292, "loss_xy": 492.4801, "loss_wh": 43.19879, "loss": 1177.34624, "grad_norm": 865.42336, "time": 0.97546}
{"mode": "train", "epoch": 161, "iter": 200, "lr": 0.001, "memory": 5572, "data_time": 0.08196, "loss_cls": 145.69064, "loss_conf": 505.96253, "loss_xy": 500.89868, "loss_wh": 44.81612, "loss": 1197.36797, "grad_norm": 878.79853, "time": 0.86075}
{"mode": "train", "epoch": 161, "iter": 300, "lr": 0.001, "memory": 5572, "data_time": 0.08167, "loss_cls": 150.64209, "loss_conf": 506.08269, "loss_xy": 500.86848, "loss_wh": 45.49562, "loss": 1203.08887, "grad_norm": 890.85835, "time": 0.83724}
{"mode": "train", "epoch": 161, "iter": 400, "lr": 0.001, "memory": 5572, "data_time": 0.40347, "loss_cls": 138.85869, "loss_conf": 482.46302, "loss_xy": 470.53182, "loss_wh": 40.89368, "loss": 1132.7472, "grad_norm": 835.73383, "time": 1.18827}
{"mode": "train", "epoch": 161, "iter": 500, "lr": 0.001, "memory": 5572, "data_time": 0.08525, "loss_cls": 137.54877, "loss_conf": 491.05236, "loss_xy": 477.2883, "loss_wh": 41.66738, "loss": 1147.55682, "grad_norm": 874.45908, "time": 0.90208}
{"mode": "train", "epoch": 161, "iter": 600, "lr": 0.001, "memory": 5603, "data_time": 0.10006, "loss_cls": 141.5963, "loss_conf": 495.99374, "loss_xy": 483.95905, "loss_wh": 43.21406, "loss": 1164.76314, "grad_norm": 854.98837, "time": 0.87027}
{"mode": "train", "epoch": 161, "iter": 700, "lr": 0.001, "memory": 5603, "data_time": 0.08042, "loss_cls": 142.90799, "loss_conf": 501.4453, "loss_xy": 491.36931, "loss_wh": 43.54654, "loss": 1179.26916, "grad_norm": 859.29139, "time": 0.89479}
{"mode": "train", "epoch": 161, "iter": 800, "lr": 0.001, "memory": 5603, "data_time": 0.10107, "loss_cls": 135.94234, "loss_conf": 481.63683, "loss_xy": 468.58937, "loss_wh": 40.9192, "loss": 1127.08774, "grad_norm": 855.19175, "time": 0.89533}
{"mode": "train", "epoch": 161, "iter": 900, "lr": 0.001, "memory": 5630, "data_time": 0.08097, "loss_cls": 146.03252, "loss_conf": 503.84448, "loss_xy": 494.09743, "loss_wh": 44.57087, "loss": 1188.54531, "grad_norm": 872.26844, "time": 0.90492}
{"mode": "train", "epoch": 161, "iter": 1000, "lr": 0.001, "memory": 5630, "data_time": 0.07724, "loss_cls": 140.72526, "loss_conf": 490.39096, "loss_xy": 482.73478, "loss_wh": 43.45077, "loss": 1157.30177, "grad_norm": 857.65189, "time": 0.74996}
{"mode": "train", "epoch": 161, "iter": 1100, "lr": 0.001, "memory": 5630, "data_time": 0.07752, "loss_cls": 142.9598, "loss_conf": 498.26939, "loss_xy": 493.91545, "loss_wh": 44.29014, "loss": 1179.43477, "grad_norm": 868.61727, "time": 0.80781}
{"mode": "train", "epoch": 161, "iter": 1200, "lr": 0.001, "memory": 5630, "data_time": 0.11144, "loss_cls": 142.52195, "loss_conf": 499.76426, "loss_xy": 495.85215, "loss_wh": 44.47356, "loss": 1182.61193, "grad_norm": 870.57327, "time": 0.98711}
{"mode": "train", "epoch": 161, "iter": 1300, "lr": 0.001, "memory": 5650, "data_time": 0.23197, "loss_cls": 146.14486, "loss_conf": 501.83429, "loss_xy": 495.92203, "loss_wh": 44.64362, "loss": 1188.5448, "grad_norm": 874.94614, "time": 1.38984}
{"mode": "train", "epoch": 161, "iter": 1400, "lr": 0.001, "memory": 5650, "data_time": 0.08052, "loss_cls": 139.87223, "loss_conf": 494.78116, "loss_xy": 481.14195, "loss_wh": 43.18185, "loss": 1158.97718, "grad_norm": 872.43151, "time": 0.80882}
{"mode": "train", "epoch": 161, "iter": 1500, "lr": 0.001, "memory": 5650, "data_time": 0.07857, "loss_cls": 140.89157, "loss_conf": 503.05241, "loss_xy": 482.89017, "loss_wh": 43.3704, "loss": 1170.20455, "grad_norm": 870.42496, "time": 0.86858}

env
Tue Jan  5 06:56:36 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  TITAN V             On   | 00000000:3D:00.0 Off |                  N/A |
| 28%   35C    P2   123W / 250W |   9011MiB / 12066MiB |     81%      Default |
+-------------------------------+----------------------+----------------------+
|   1  TITAN V             On   | 00000000:3F:00.0 Off |                  N/A |
| 28%   35C    P2    40W / 250W |   9941MiB / 12066MiB |     82%      Default |
+-------------------------------+----------------------+----------------------+
|   2  TITAN V             On   | 00000000:40:00.0 Off |                  N/A |
| 28%   34C    P2    41W / 250W |  10431MiB / 12066MiB |     44%      Default |
+-------------------------------+----------------------+----------------------+
|   3  TITAN V             On   | 00000000:43:00.0 Off |                  N/A |
| 28%   33C    P2    39W / 250W |   9289MiB / 12066MiB |     78%      Default |
+-------------------------------+----------------------+----------------------+
|   4  TITAN V             On   | 00000000:44:00.0 Off |                  N/A |
| 28%   34C    P2    39W / 250W |  10741MiB / 12066MiB |     80%      Default |
+-------------------------------+----------------------+----------------------+
|   5  TITAN V             On   | 00000000:45:00.0 Off |                  N/A |
| 28%   36C    P2    37W / 250W |  12051MiB / 12066MiB |     73%      Default |
+-------------------------------+----------------------+----------------------+

**Comment by ZwwWayne** [comment]
Hi @yy18dlz ，
Could you provide a specific config for use to reproduce? And are the time of both codebases tested  on the similar machine? We have not checked the speed of YOLO for now.

**Comment by yy18dlz** [comment]
@ZwwWayne 
Sorry,for some reason I can't provide specific config. I can tell you what I've changed.
1. Writing a customize dataset for my  project
2. Writing the Resize func on data augmen piplines that like darknet letter box resize
3. For create model, I chose .cfg,  just like darknet style, I think it's more flexible.
So, at first, I think maybe the data load cost the most time, But in log, we can found the most of the time is spent on training inference.
